develop/tutorials/articles/01-introduction-to-liferay-development/01-fundamentals.markdown

```diff
@@ -81,37 +80,90 @@ taking the server down.
 
 +$$$
 
-Note: In the OSGi nomenclature, modules are sometimes called OSGi bundles, but
-we will avoid that terminology to avoid confusing it with the concept of a
-Liferay bundle, which is a distribution of Liferay bundled with an app server. 
+Note: In the OSGi nomenclature, modules are sometimes called OSGi bundles. An
+attempt has been made to be consistent and only use the word *modules* to refer
+to modules, but if a *bundle* has slipped in here and there, just know they're
+the same thing. 
 
 $$$
 
-Modules may have one or more *components*. Creating a component can be as easy
-as writing a Java class with the `@Component` annotation. In software
-engineering terms, a component is the smallest building block of a larger
-application, and that application is itself made up of many small components.
-This makes it easier to develop an application because you only have to deal
-with small, well-defined, bite-sized chunks of code at a time. 
+## Components
 
-<!-- Why should I have components? What do I need them for? 
-Components can also be extension points. A dynamic extension points module. -->
+Modules may contain one or more *components*. A component is a POJO whose
+life cycle is managed by the container. In Liferay, a component is generally part
+of the Declarative Services framework. Creating a component can be as easy as
+writing a Java class with the `@Component` annotation. In software engineering
+terms, a component is the smallest building block of a larger application, and
+that application is itself made up of many small components.  This makes it
+easier to develop an application because you only have to deal with small,
+well-defined, bite-sized chunks of code at a time. 
 
 Components are managed by a component container, which keeps track of the
-components that are installed and active. Components offer services, and
-it's the container's job to manage the available services by providing
-components the services they need. It does this through a robust dependency
-management system that's handled automatically at runtime. 
-
-<!-- Mention dependency injection and the fact that it's better than Spring
-because it's dynamic. -->
+components that are installed and active. In this way, components have these
+benefits: 
 
-As you can see, many components running inside the system form something of an
-ecosystem of complementary services. @product@ ships with many components that are
-ready to use out of the box, and of course you can develop your own. You can
-write components to provide new services or to override existing services with
-your own implementation, and the container manages it all. Liferay is an
-exciting platform that empowers developers to be more productive. 
+- The container can start and stop components. 
+- A component *may* publish itself as an OSGi service, but not all components
+    are services. 
```

Comment:
I'm not sure if this is true. I think that all Components are automatically registered as OSGi services. Where did you get this from?